app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/add_tools/+page.svelte

```diff
@@ -0,0 +1,176 @@
+    const headersObj: Record<string, string> = {}
+
+    for (const header of headers) {
+      if (header.key.trim() && header.value.trim()) {
+        headersObj[header.key.trim()] = header.value.trim()
+      }
+    }
+
+    return headersObj
+  }
+
+  async function connect_remote_mcp() {
+    try {
+      error = null
+      submitting = true
+
+      // Validate required fields
+      if (!name.trim()) {
+        throw new Error("Name is required")
+      }
+      if (!server_url.trim()) {
+        throw new Error("Server URL is required")
+      }
+      if (headers.length > 0) {
+        for (const header of headers) {
+          if (!header.key.trim()) {
+            throw new Error("Header name is required")
+          }
+          if (!header.value.trim()) {
+            throw new Error("Header value is required")
+          }
+        }
+      }
+
+      const headersObj = buildHeadersObject()
+
+      const { data, error: api_error } = await client.POST(
+        "/api/projects/{project_id}/connect_remote_MCP",
+        {
+          params: {
+            path: {
+              project_id: $page.params.project_id,
+            },
+          },
+          body: {
+            name: name.trim(),
+            server_url: server_url.trim(),
+            headers: headersObj,
+            description: description.trim() || null,
+          },
+        },
+      )
+
+      if (api_error) {
+        throw api_error
+      }
+
+      if (data?.id) {
+        // Navigate to the tools page for the created tool
+        goto(
+          `/settings/manage_tools/${$page.params.project_id}/tools/${data.id}`,
+        )
+      }
+    } catch (e) {
+      error = createKilnError(e)
+    } finally {
+      submitting = false
+    }
+  }
+</script>
+
+<AppPage
+  title="Connect Remote MCP Server"
+  sub_subtitle="Connect to a remote Model Context Protocol (MCP) server to add external
+        tools to your project."
+>
+  <div class="max-w-2xl">
+    <FormContainer
+      submit_label="Connect"
+      on:submit={connect_remote_mcp}
+      bind:error
+      bind:submitting
+    >
+      <FormElement label="Name" id="name" bind:value={name} max_length={120} />
+
+      <FormElement
+        label="Description"
+        inputType="textarea"
+        id="mcp_description"
+        description="A description of this MCP server for your reference."
+        optional={true}
+        bind:value={description}
+      />
+
+      <FormElement
+        label="Server URL"
+        id="mcp_server_url"
+        description="The URL of the remote MCP server"
+        placeholder="https://example.com/mcp"
+        bind:value={server_url}
+      />
+
+      <!-- Headers section -->
+      <FormElement
+        inputType="header_only"
+        label="Headers"
+        id="headers_section"
+        description="If the documentation for the server you're adding requires custom headers, enter them here."
+        optional={true}
```

Comment:
optional=false here? Optional label doesn't quite apply to a list UI that allows zero items already.